server/src/main/java/org/loomdev/loom/world/WorldImpl.java

```diff
@@ -84,16 +86,14 @@ public Chunk getChunk(Vector3i chunkpos) {
     }
 
     @Override
-    @Nullable
-    public <T extends Entity> T spawnEntity(@NotNull EntityType<T> type, @NotNull Location location) {
-        var mcEntity = Registry.ENTITY_TYPE.get(new ResourceLocation(type.getKey().toString())).create(getMinecraftWorld());
-        if (mcEntity == null) {
-            return null;
-        }
-
-        mcEntity.moveTo(location.getX(), location.getY(), location.getZ(), location.getYaw(), location.getPitch());
-        getMinecraftWorld().addFreshEntity(mcEntity);
-        return (T) mcEntity.getLoomEntity();
+    @NotNull
+    public <T extends Entity> Optional<T> spawnEntity(@NotNull EntityType<T> type, @NotNull Location location) {
+        return Optional.ofNullable(Registry.ENTITY_TYPE.get(new ResourceLocation(type.getKey().toString())).create(getMinecraftWorld()))
+                .map(entity -> {
+                    entity.moveTo(location.getX(), location.getY(), location.getZ(), location.getYaw(), location.getPitch());
+                    getMinecraftWorld().addFreshEntity(entity);
+                    return (T) entity.getLoomEntity();
```

Comment:
The `moveTo` and `addFreshEntity` calls should be in a `peek(entity -> ...)` function not the map. (But keep them together 😄)

Reply:
Sorry, my mistake. This is an optional not a Stream. 😅